src/wallet3/output_selection/output_selection.cpp

```diff
@@ -0,0 +1,49 @@
+#include "output_selection.hpp"
+
+namespace wallet
+{
+  std::vector<Output>
+  SelectOutputs(const std::vector<Output>& available_outputs, int64_t amount)
+  {
+
+    // Check that we actually have enough in the outputs to build this transaction. Fail early
```

Comment:
This could actually be done before we even get to SelectOutputs, as a db query.  Not sure if better that way though.

Reply:
I'm bout to push an different selection algorithm that uses standard library distributions. Figured the algorithm will have to get smarter and bring in a fair bit of extra that sqlite cant do